service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobResult.java

```diff
@@ -155,7 +155,8 @@
         TwinParser twinParser = jobsResponseParser.getUpdateTwin();
         if(twinParser != null)
         {
-            this.updateTwin = new DeviceTwinDevice(twinParser.getDeviceId());
+            this.updateTwin = twinParser.getDeviceId() == null || twinParser.getDeviceId().isEmpty() ?
+                new DeviceTwinDevice() : new DeviceTwinDevice(twinParser.getDeviceId());
```

Comment:
Please add a test case for empty device ID in the corresponding test.

Reply:
Test suite for this is located https://github.com/Azure/azure-iot-sdk-java/blob/master/service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/jobs/JobResultTest.java

Reply:
@prmathur-microsoft , a test case of using nullable deviceId has been added in the commit. please help review it. Thanks!